### ISSUE TITLE
www.jsdelivr.com is being censored in Bulgaria

### ISSUE THREAD

**Issue by tmcdos** [jsdelivr/jsdelivr#18082]
Are you aware that https://www.jsdelivr.com is banned in Bulgaria by the government agency for gambling ?
When I try to open the website I get this
![jsdelivr](https://user-images.githubusercontent.com/19253115/44142806-15782b5a-a08a-11e8-8804-1dccfac40f71.png)
The actual page content is
```
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
    <head>
      <meta http-equiv="Content-Type" content="text/html;charset=Utf-8" >
      <meta http-equiv="pragma" content="no-cache" >
      <meta http-equiv="Cache-control" content="no-cache" >
    </head>
<body bgcolor="#FFFFFF">
<center>
<font face="verdana">
Търсената от Вас страница е с неразрешено съдържание и достъпът Ви до нея е ограничен.
<br />
<br />
За повече информация проверете на: <br />
Държавна Комисия по Хазарта - 
<a href="http://www.dkh.minfin.bg/"> www.dkh.minfin.bg </a> </font><br />

</body>
</html>

```
This content claims to be coming from 104.31.90.167 - and 104.31.91.167 seems to refuse connections on port 443 (from Bulgaria).

**Comment by jimaek** [comment]
That is very strange. Do you know if the government has some kind of website with blocked IPs to check what was exactly blocked?

**Comment by tmcdos** [comment]
Nope, there is no such list. On the website (http://www.dkh.minfin.bg/bg/pubs/2) they have a long list with court decisions and decisions of the agency itself (on which the court ones are based - http://www.dkh.minfin.bg/bg/pubs/4) - I have reviewed them all and www.jsdelivr.com was not mentioned anywhere (unfortunately the English version of the site is very incomplete). However, when I complained to a major ISP they vaguely replied like this:
![shot-2](https://user-images.githubusercontent.com/19253115/44144998-77c2a65c-a092-11e8-9a2d-11cc0312f4a9.png)
I checked the court decision from June 17 2013 they cited but www.jsdelivr.com was not mentioned there.
So the ISP claims they obey a court decision, I was unable to find www.jsdelivr.com mentioned in any of the court decisions, the government agency does not respond by email, and the fact is that the website is not accessible from Bulgaria (at least through the few major ISPs).

**Comment by jimaek** [comment]
I believe the issue is the IP owned by Cloudflare. I am talking with their support to replace our IPs with clean ones.

**Comment by tmcdos** [comment]
The ISP is showing they do not give a shit. Their latest response is
![shot-1](https://user-images.githubusercontent.com/19253115/44299373-29696a00-a2fd-11e8-9bc1-15dbe5e75f4b.png)

Translated to English:

Greetings

Many websites use the CDN network of www.jsdelivr.com to store their JavaScript libraries,
However the government agency for gambling has banned this website in Buglaria.
When I asked why through the live chat on your web portal the operator insisted for my name, ID number, address, etc - otherwise my complain can not be accepted. I provided the number of my contract since the contract contains all this data already - but this did not help.
As a result my rights are violated in 3 ways:
a) my complain was declined by the support staff
b) I can not access www.jsdelivr.com even though it has nothing to do with betting
c) my right to access public government information (specifically - the ID of the relevant court order) was refused

Hellow
A1 obeys the court order from June 17 2013 by the Sofia's regional court. This order obligates all Bulgarian ISPs to limit the access to a list of gambling web sites when their owners do not have a proper license. In this particular case A1 simply executes the court order. Legal activities to dispute it can be performed only by the owners of the corresponding web sites.

Hi
I looked at the court order that you mentioned and www.jsdelivr.com was not listed there - only 21 betting websites. Are you sure that you have not mistakenly banned this website even though it was not listed in the court order? The court order is based on the  request 674 from June 11 2013 by the government agency for gambling but www.jsdelivr.com was not listed there, too.

![att00001](https://user-images.githubusercontent.com/19253115/44299545-02149c00-a301-11e8-8582-54cb5c835eae.png)
![att00002](https://user-images.githubusercontent.com/19253115/44299546-05a82300-a301-11e8-9e57-36e6d73fd9a9.png)

Hellow
I count on your understanding that we have already answered your questions in the previous reply. Information about which websites have a proper license by the agency for gambling can be found at http://www.dkh.minfin.bg/

---------------------------------------------------------
You are probably right that the issue might be solved by changing the IPs - but we have yet to see.

**Comment by MartinKolarik** [comment]
@tmcdos could you please check if https://cdn.jsdelivr.net/npm/jquery/ works for you and what IP it resolves to?

Additionally, if you have a shell with `curl` at hand, could you share the results of following commands?

```
curl 104.31.90.167
curl -I 104.31.90.167
curl -I -H 'Host: www.jsdelivr.com' 104.31.90.167

curl https://www.jsdelivr.com
curl -I https://www.jsdelivr.com
```

**Comment by tmcdos** [comment]
Yes, it works - resolved to 104.16.89.20
![shot-1](https://user-images.githubusercontent.com/19253115/44307473-cbe02680-a3ab-11e8-8982-6edbd2535004.png)
![shot-2](https://user-images.githubusercontent.com/19253115/44307474-cedb1700-a3ab-11e8-8c0f-6595bcf97b61.png)

```
[root@fedora25: /tmp] curl 104.31.90.167
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
    <head>
      <meta http-equiv="Content-Type" content="text/html;charset=Utf-8" >
      <meta http-equiv="pragma" content="no-cache" >
      <meta http-equiv="Cache-control" content="no-cache" >
    </head>
<body bgcolor="#FFFFFF">
<center>
<font face="verdana">
Търсената от Вас страница е с неразрешено съдържание и достъпът Ви до нея е ограничен.
<br />
<br />
За повече информация проверете на: <br />
Държавна Комисия по Хазарта -
<a href="http://www.dkh.minfin.bg/"> www.dkh.minfin.bg </a> </font><br />

</body>
</html>
[root@fedora25: /tmp]
```
```
[root@fedora25: /tmp] curl -I 104.31.90.167
HTTP/1.1 200 OK
Server: nginx/0.7.67
Date: Sun, 19 Aug 2018 09:32:27 GMT
Content-Type: text/html
Content-Length: 840
Last-Modified: Tue, 29 Sep 2015 15:04:48 GMT
Accept-Ranges: bytes

[root@fedora25: /tmp]
```
```
[root@fedora25: /tmp] curl -I -H 'Host: www.jsdelivr.com' 104.31.90.167
HTTP/1.1 200 OK
Server: nginx/0.7.67
Date: Sun, 19 Aug 2018 09:33:00 GMT
Content-Type: text/html
Content-Length: 840
Last-Modified: Tue, 29 Sep 2015 15:04:48 GMT
Accept-Ranges: bytes

[root@fedora25: /tmp]
```
```
[root@fedora25: /tmp] curl https://www.jsdelivr.com
curl: (7) Failed to connect to www.jsdelivr.com port 443: Connection refused
[root@fedora25: /tmp] curl -I https://www.jsdelivr.com
curl: (7) Failed to connect to www.jsdelivr.com port 443: Connection refused
[root@fedora25: /tmp]
```

**Comment by MartinKolarik** [comment]
Thanks. This is indeed the CF's IP being blocked and not our domain directly. The IPs are shared by many domains so most likely one of the gambling sites was on the same IP. As @jimaek said we're talking with CF to fix this.
